### ISSUE TITLE
How to pass entire JWT to backend API

### ISSUE THREAD

**Issue by molinab297-unisys** [wso2/product-microgateway#1335]
Hello, I just have a question. According to the docs [here](https://docs.wso2.com/display/MG310/Secure+APIs+using+JWT+-+Self+Contained+JWT), it says I can pass a single claim value to my backend. Is there a way that I can forward the _entire_ JWT to my backend? I want to be able to log several things about the client in my backend (like username, email, etc), in addition to getting all of their scopes. Do I really only have one single claim to work with?

It seems like I can do this with the API Manager (see [here](https://docs.wso2.com/display/AM260/Passing+Enduser+Attributes+to+the+Backend+Using+JWT)). Can I achieve this with the Microgateway?

**Comment by Rajith90** [comment]
Hi @molinab297-unisys ,
Micro gateway can send the jwt coming in the  authorisation header to the back end as it is. This can be achived by using the following config

```toml
[authConfig]
  # Remove authorization header from the backend request
  removeAuthHeaderFromOutMessage = false
```
And also If you are using API Manager 3.0.0 it sends the back end jwt as a single claim in the authorisation header.  That claim is "backendJwt" claim. API Manager sends base 64 encoded jwt to be sent to the back end in this claim in the jwt coming in the authorisation header.

And microgateway also can generate a jwt to be sent to the back end by looking at the claims in the incoming auth jwt(similar to API manager 3.1.0). This feature was added as a wum update for microgateway 3.1.0. If you have a valid subscription for microgateway you can use the wum updated pack. And this feature is also available in 3.2.0- beta version as well. Still 3.2.0 GA release is not done.

**Comment by molinab297-unisys** [comment]
@Rajith90 Thank you for your help. That worked for me.
